connector/openshift/openshift.go

```diff
@@ -109,9 +119,30 @@ func (c *Config) Open(id string, logger log.Logger) (conn connector.Connector, e
 		Scopes:      []string{"user:info"},
 		RedirectURL: c.RedirectURI,
 	}
+
+	if err := openshiftConnector.queryOAuth2Endpoints(); err != nil {
+		cancel()
+		return nil, fmt.Errorf("failed to query oauth2 endpoint: %v",
+			err)
+	}
+
 	return &openshiftConnector, nil
 }
 
+func (c *openshiftConnector) queryOAuth2Endpoints() error {
```

Comment:
`query` is not a very descriptive name. Maybe `checkOAuth2EndpointsAvailability`?

Reply:
done within #2